fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/query/plan/cascades/WithPrimaryKeyMatchCandidate.java

```diff
@@ -36,16 +33,6 @@ public interface WithPrimaryKeyMatchCandidate extends MatchCandidate {
     @Nonnull
     Optional<List<Value>> getPrimaryKeyValuesMaybe();
 
-    @Nonnull
```

Comment:
I think moving this up is the correct thing to do.

Reply:
Yeah I moved it to `MatchCandidate` as I thought it is not really relevant to `WithPrimaryKeyMatchCandidate` trait.